### PR TITLE
image_dataset.py doesn't work if an input image is gray scale

### PR DESCRIPTION
**Pull request by yasunorikudo**
If the image is gray scale, an error, "ValueError: axes don't match array" will occur in line 123.

### CODE REVIEW

**Comment by delta2323** [comment]
Thank you for sending the PR. Could you change the title more explanatory one?

**Comment by yasunorikudo** [comment]
I'm sorry for your inconvenience. I changed the title.

File: chainer/datasets/image_dataset.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def get_example(self, i):
         path, int_label = self._pairs[i]
         full_path = os.path.join(self._root, path)
         with Image.open(full_path) as f:
-            image = numpy.asarray(f, dtype=self._dtype)
+            image = numpy.asarray(f.convert('RGB'), dtype=self._dtype)
```
Comment (zori) [inline chainer/datasets/image_dataset.py]: For an input grey-scale image (H, W), this solution will output a (3, H, W) ndarray. However as stated here https://github.com/pfnet/chainer/blob/master/chainer/datasets/image_dataset.py#L24 we'd like to output a (1, H, W) for a grey-scale input. How about `image = image[:, :, numpy.newaxis]` if the input is 2D?

**Comment by zori** [comment]
Please, also fix `ImageDataset` after this line https://github.com/pfnet/chainer/blob/master/chainer/datasets/image_dataset.py#L61 (analogous code).

**Comment by zori** [comment]
Could you please write a minimal test demonstrating the fix, using RGB and greyscale data in chainer/chainer/datasets/image_dataset.py?

**Comment by yasunorikudo** [comment]
I also changed docs in examples/imagenet/train_imagenet.py because ImageNet includes some greyscale images.
If we don't convert them to color images, channel size doesn't match with all images.

File: chainer/datasets/image_dataset.py [hunk 2]
```diff
@@ -59,6 +59,8 @@ def get_example(self, i):
         path = os.path.join(self._root, self._paths[i])
         with Image.open(path) as f:
             image = numpy.asarray(f, dtype=self._dtype)
+        if len(image.shape) == 2:
```
Comment (zori) [inline chainer/datasets/image_dataset.py]: I believe `if image.ndim == 2` is the more common, please use it for readability. Could you also comment `# image is greyscale` on the next line?

File: chainer/datasets/image_dataset.py [hunk 3]
```diff
@@ -119,6 +121,8 @@ def get_example(self, i):
         full_path = os.path.join(self._root, path)
         with Image.open(full_path) as f:
             image = numpy.asarray(f, dtype=self._dtype)
+        if len(image.shape) == 2:
```
Comment (zori) [inline chainer/datasets/image_dataset.py]: ditto: check for greyscale with `image.ndim`

**Comment by zori** [comment]
Great, thank you. LGTM except the above two comments. `image_dataset.py` is not tested yet, so perhaps it's alright to leave it like that for the moment.

**Comment by yasunorikudo** [comment]
Thank you for your advices. I fixed the code.

**Comment by unnonouno** [comment]
Do you have a minimum code to test it?

**Comment by unnonouno** [comment]
`shape` of images are different between colored images and greyscale images. Is it acceptable?

**Comment by unnonouno** [comment]
On offline discussion, we concluded that a user is responsible to convert colored/greyscale images.
LGTM.

**Pull request merged**